VimeoUpload/Upload/Controllers/VideoDeletionManager.swift

```diff
@@ -131,28 +134,29 @@ public class VideoDeletionManager: NSObject
     
     // MARK: Archiving
     
-    private func loadDeletions() -> [VideoUri: Int]?
+    private func loadDeletions(withMigrator migrator: ArchiveMigrating?) -> [VideoUri: Int]
     {
         guard self.shouldLoadArchive == true else
         {
-            return nil
+            return [:]
         }
         
-        if let deletions = self.archiver.loadObject(for: type(of: self).DeletionsArchiveKey) as? [VideoUri: Int]
+        let relativeFolderPath = VideoDeletionManager.DeletionsArchiveKey + "/" + VideoDeletionManager.DeletionsArchiveKey
+        
+        guard let retries = ArchiveDataLoader.loadData(relativeFolderPath: relativeFolderPath,
+                                                       archiver: self.archiver,
+                                                       key: VideoDeletionManager.DeletionsArchiveKey,
+                                                       migrator: migrator) as? [VideoUri: Int]
+        else
         {
-            return deletions
+            return [:]
```

Comment:
Would it make sense to have this method return an optional instead? Or can the empty dictionary still be used?

Reply:
We still need an empty dictionary here because we still need to save to archive. With an optional property, we won't be able to save anything.